documents/entity/service.go

```diff
@@ -0,0 +1,224 @@
+package entity
+
+import (
+	"context"
+
+	"github.com/centrifuge/centrifuge-protobufs/gen/go/coredocument"
+	"github.com/centrifuge/go-centrifuge/contextutil"
+	"github.com/centrifuge/go-centrifuge/documents"
+	"github.com/centrifuge/go-centrifuge/errors"
+	cliententitypb "github.com/centrifuge/go-centrifuge/protobufs/gen/go/entity"
+	"github.com/centrifuge/go-centrifuge/queue"
+	"github.com/centrifuge/go-centrifuge/transactions"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+)
+
+// Service defines specific functions for entity
+type Service interface {
+	documents.Service
+
+	// DeriverFromPayload derives Entity from clientPayload
+	DeriveFromCreatePayload(ctx context.Context, payload *cliententitypb.EntityCreatePayload) (documents.Model, error)
+
+	// DeriveFromUpdatePayload derives entity model from update payload
+	DeriveFromUpdatePayload(ctx context.Context, payload *cliententitypb.EntityUpdatePayload) (documents.Model, error)
+
+	// DeriveEntityData returns the entity data as client data
+	DeriveEntityData(inv documents.Model) (*cliententitypb.EntityData, error)
```

Comment:
just a nitpick, but maybe rename the `inv` parameter because you are deriving it from an Entity document model, could be confusing.

Reply:
👍